Doc/library/unittest.mock.rst

```diff
@@ -1343,15 +1343,24 @@ patch.dict
     decorator. When used as a class decorator :func:`patch.dict` honours
     ``patch.TEST_PREFIX`` for choosing which methods to wrap.
 
+    .. versionchanged:: 3.8
+
+        :func:`patch.dict` now returns the patched dictionary when used as a context
+        manager.
+
 :func:`patch.dict` can be used to add members to a dictionary, or simply let a test
 change a dictionary, and ensure the dictionary is restored when the test
 ends.
 
     >>> foo = {}
-    >>> with patch.dict(foo, {'newkey': 'newvalue'}):
+    >>> with patch.dict(foo, {'newkey': 'newvalue'}) as patched_foo:
     ...     assert foo == {'newkey': 'newvalue'}
+    ...     assert patched_foo == {'newkey': 'newvalue'}
+    ...     patched_foo['spam'] = 'eggs'
+    ...     assert patched_foo['spam'] == 'eggs'
```

Comment:
Were these last two lines added after my request?

What I meant is: show what the two dicts (original, returned by the context manager) look like after the with block.

Reply:
Nope, for that comment I added `assert patched_foo == {}` after the context manager.

Reply:
Right I missed that in the PR view!

I don’t really get what these two lines bring.  Of course if I add something to a dict on line 4 I will see it by looking at the dict on line 5?

Did you mean: add an item to foo, then show patched_foo is the same?

Reply:
It is just "working with the dict". We can totally remove these two lines if you want. It is probably simpler/clearer indeed. Do you want me to remove them?

Reply:
Ah I think it’s useful to have one line!  Sometimes the call to `patch.dict` has all the keys needed, sometimes you add some in the context manager body. Maybe add a comment `# You can add, update or delete keys of foo or patched_foo`